In [1]:
import numpy as np
from scipy import stats
import scipy.stats
import pandas as pd
import statsmodels.api as sm
import matplotlib.pyplot as plt
import seaborn as sns
from statsmodels.stats.proportion import proportions_ztest
from scipy.stats import shapiro

# p.131

In [2]:
df1 = pd.DataFrame({'sample': [85.0, 79.0, 79.1, 79.9, 81.6, 78.6, 85.4, 83.4, 78.1, 79.2]})
t_result1 = stats.ttest_1samp(df1, 76.7)
t1,p1 = t_result1.statistic.round(3), t_result1.pvalue.round(3)
print("1-sample t-test")
print("t 통계량: {}".format(t1))
print("p-value: {}".format(p1))

1-sample t-test
t 통계량: [4.879]
p-value: [0.001]


유의수준 5%에서 p-value 값이 0.001이므로 귀무가설이 기각된다.

# p.139

In [3]:
#등분산성 검정/ 등분산성이 있다(귀무가설 )/ 
df1 = pd.DataFrame({'A': [1.883, 1.715, 1.799, 1.768, 1.711, 1.832, 1.427, 1.344]})
df2 = pd.DataFrame({'B': [1.435, 1.572, 1.486, 1.511, 1.457, 1.548, 1.404, 1.883]})
stats.levene(df1['A'], df2['B'])

LeveneResult(statistic=0.4408416728536036, pvalue=0.5175025061092301)

등분산 검정 결과 p-value가 0.51이므로 귀무가설이 기각 되지 않는다. 즉, 정규성이 존재한다.

In [4]:
t_result = stats.ttest_ind(df1,df2)
t,p = t_result.statistic.round(3), t_result.pvalue.round(3)
print("2-sample t-test")
print("t: {}".format(t))
print("p: {}".format(p))

2-sample t-test
t: [1.7]
p: [0.111]


유의수준 5%에서 p-value 값이 0.111이므로 귀무가설이 기각 되지 않는다.

# p.142

In [6]:
df1 = pd.DataFrame({'sample': [720, 589, 780, 648, 720, 589, 780, 648, 780, 648]})
df2 = pd.DataFrame({'sample': [710, 580, 787, 712, 750, 600, 782, 670, 790, 680]})
t_result = stats.ttest_rel(df1,df2)
t,p = t_result.statistic.round(3), t_result.pvalue.round(3)
print("Paired t-test")
print("t: {}".format(t))
print("p: {}".format(p))

Paired t-test
t: [-2.266]
p: [0.05]


유의수준 5%에서 p-value 값이 0.05이므로 귀무가설이 기각 되지 않는다.
(기각도 충분히 할 수는 있지만 보수적으로 접근하고자 했습니다.)

# p.148

In [7]:
count = 40
n_obs = 100
value = 0.5

stat, pval = proportions_ztest(count, n_obs, value)
print("1 Proportion test")
print("z: {0:0.3f}".format(stat))
print("p: {0:0.3f}".format(pval))

1 Proportion test
z: -2.041
p: 0.041


유의수준 5%에서 p-value 값이 0.041이므로 귀무가설이 기각된다.

# p.151

In [8]:
count = np.array([14,5])
nobs = np.array([1200, 1200])

stat, pval = proportions_ztest(count, nobs)
print("2 Proportion test")
print("z: {0:0.3f}".format(stat))
print("p: {0:0.3f}".format(pval))

2 Proportion test
z: 2.073
p: 0.038


유의수준 5%에서 p-value 값이 0.038이므로 귀무가설이 기각된다.

# p.159

In [9]:
df = pd.DataFrame({'A/S': [18, 8, 4, 4, 3, 3], '가격':[1,2,1,1,1,25], '성능': [8,14,3,2,3,8], '확장성': [7,5,4,3,1,10], '디자인': [10,5,9,2,1,2], '안정성': [9,9,5,7,1,1], '기능성': [10,4,4,3,1,7]})
chi, pval, dof, expected = stats.chi2_contingency(df.T)
print("chi-square test")
print("chisq: {0:0.3f}".format(chi))
print("p: {0:0.3f}".format(pval))
print("degree of freedom: {}".format(dof))
print("expected value: \n{}".format(expected.round(3)))

chi-square test
chisq: 98.274
p: 0.000
degree of freedom: 30
expected value: 
[[11.004  8.21   5.24   3.843  1.921  9.782]
 [ 8.528  6.362  4.061  2.978  1.489  7.581]
 [10.454  7.799  4.978  3.651  1.825  9.293]
 [ 8.253  6.157  3.93   2.882  1.441  7.336]
 [ 7.978  5.952  3.799  2.786  1.393  7.092]
 [ 8.803  6.568  4.192  3.074  1.537  7.825]
 [ 7.978  5.952  3.799  2.786  1.393  7.092]]


유의수준 5%에서 p-value 값이 0.000이므로 귀무가설이 기각된다.

# p.195

In [10]:
strength = [37.6,38.6,37.2,36.4,38.6,39,37.2,36.1,35.0,37.1,36.9,37.5,36.3,38.1,39,36.9,36.8,37.6,33,33.5]
bakingtime = [0.798922,0.84902,0.816163,0.758266,0.815894,0.807477,0.809068,0.7664,0.742243,0.751118,0.740629,0.751495,0.738863,0.827428,0.818172,0.740939,0.757128,0.804695,0.695121,0.735377]
strength = sm.add_constant(strength)
model = sm.OLS(bakingtime, strength)
result = model.fit()
print(result.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.650
Model:                            OLS   Adj. R-squared:                  0.631
Method:                 Least Squares   F-statistic:                     33.43
Date:                Wed, 02 Sep 2020   Prob (F-statistic):           1.77e-05
Time:                        01:53:13   Log-Likelihood:                 46.635
No. Observations:                  20   AIC:                            -89.27
Df Residuals:                      18   BIC:                            -87.28
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0225      0.130      0.172      0.8

회귀식은 bakingtime = 0.0225 + 0.0204 strength. 설명력은 R-Sq 65%이다.

# p.198

In [14]:
df = pd.read_csv("correlation.csv", engine = 'python')
from sklearn.linear_model import LinearRegression
import statsmodels.formula.api as smf
model = smf.ols(formula = "strength ~ Oven_TMP", data = df)
result = model.fit()
print(result.summary())

                            OLS Regression Results                            
Dep. Variable:               strength   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                 -0.011
Method:                 Least Squares   F-statistic:                    0.1136
Date:                Wed, 02 Sep 2020   Prob (F-statistic):              0.737
Time:                        02:01:31   Log-Likelihood:                -223.88
No. Observations:                  86   AIC:                             451.8
Df Residuals:                      84   BIC:                             456.7
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     30.4190      5.271      5.771      0.0

회귀식은 strength = 30.4190 + 0.0056 Oven_TMP. 설명력은 R-Sq 0.1%이다.